### PR TITLE
Fix C++ multi-package UP release

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

**Review by Jinming-Hu — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-cpp-release.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ parameters:
 stages:
   - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
     - ${{ each artifact in parameters.Artifacts }}:
-      - stage: Release_${{artifact.safeName}}
+      - stage: Release_${{ replace(artifact.Name, '-', '_') }}
```
Comment (mitchdenny) [inline eng/pipelines/templates/stages/archetype-cpp-release.yml:8]: nit: lowercase n?

**Review by mitchdenny — APPROVED** [review]
You might find that the repetition of replace(name) becomes so annoying that you go back to putting safe-name in ;)

File: eng/pipelines/templates/stages/archetype-cpp-release.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-cpp-release.yml:8]: @mitchdenny is this a potential fix for needing the safeName in other languages?

Comment (mitchdenny) [reply eng/pipelines/templates/stages/archetype-cpp-release.yml:8]: Yes .... I was thinking the same thing.

**Pull request merged**